### PR TITLE
[AppSignals] Explicitly set the GCCollectionInterval for MetricsLimiter

### PR DESCRIPTION
**Pull request by sky333999**
# Description of the issue
When not explicitly set, the current validate logic doesnt seem to get invoked before the start of the component, so it doesnt respect the default GC collection interval causing an infinite loop leading to CPU maxing out.

# Description of changes
Explicitly set the GC collection interval when initializing the config.

# License
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

# Tests
Manually tested on an EKS cluster that CPU spike goes away.

# Requirements
_Before commit the code, please do the following steps._
1. Run `make fmt` and `make fmt-sh`
2. Run `make lint`

### CODE REVIEW

**Review by jefchien — DISMISSED** [review]

**Review by lisguo — DISMISSED** [review]

File: plugins/processors/awsappsignals/config/config.go [hunk 1]
```diff
@@ -34,10 +34,11 @@ const (
...
 	return &LimiterConfig{
-		Threshold:         DefaultThreshold,
-		Disabled:          false,
-		LogDroppedMetrics: false,
-		RotationInterval:  DefaultRotationInterval,
+		Threshold:                 DefaultThreshold,
+		Disabled:                  false,
+		LogDroppedMetrics:         false,
+		RotationInterval:          DefaultRotationInterval,
+		GarbageCollectionInterval: DefaultGCInterval,
```
Comment (bjrara) [inline plugins/processors/awsappsignals/config/config.go:41]: Hmmm, interesting, I thought`GarbageCollectionInterval` will be assigned with `DefaultGCInterval` in the `Validate()` if no custom value is provided.

Does the test shows that after assigning a default value, the CPU usage drops?

**Review by mxiamxia — APPROVED** [review]

File: translator/translate/otel/processor/awsappsignals/testdata/invalidRulesConfig.json [hunk 2]
```diff
@@ -6,7 +6,8 @@
         "limiter": {
           "drop_threshold": 20,
           "log_dropped_metrics": true,
-          "rotation_interval": "10m"
+          "rotation_interval": "10m",
+          "garbage_collection_interval": "10m"
```
Comment (jefchien) [inline translator/translate/otel/processor/awsappsignals/testdata/invalidRulesConfig.json:10]: Was this added to the schema.json?

Comment (lisguo) [reply translator/translate/otel/processor/awsappsignals/testdata/invalidRulesConfig.json:10]: yeah, also not sure if this was meant to be configurable by customers

Comment (sky333999) [reply translator/translate/otel/processor/awsappsignals/testdata/invalidRulesConfig.json:10]: Ppl can use it even if its not in our schema afaik depending on additonalPros..
None of the metric limiter props are in our schema.json - we should revisit.
For now, I just added to all the places `rotation_interval` is present.

File: plugins/processors/awsappsignals/config/config.go (hunk 1 above)
Comment (jefchien) [reply plugins/processors/awsappsignals/config/config.go:41]: Do you mean the config `Validate()` function? That's called by the OTEL collector on start up (https://github.com/open-telemetry/opentelemetry-collector/blob/8990be35990b9833ad98d9bc254667560b7a9951/otelcol/collector.go#L182). We really shouldn't be relying on the Validate function to set defaults in the config.

**Review by jefchien — APPROVED** [review]

**Review by lisguo — APPROVED** [review]

**Review by mxiamxia — APPROVED** [review]

**Pull request merged**